### DISCUSSION TITLE
v8.2.1-beta Added missing Assistant features

### DISCUSSION THREAD

**Discussion by kayhantolga** [betalgo/openai#547] [Announcements]
https://github.com/betalgo/openai/releases/tag/v8.2.1-beta
I have added missing features, I didn't have enough time to test new implementations yet, but I shared so if it works you can use if it doesn't you let me know and I can fix or you can send a pr.
Note: I haven't check what is deprecated in assistant v1 yet.

Complex request and response objects, coupled with unclear documentation, are causing progress to be quite slow. I attempted to build an auto-generator from OpenAI's official OpenAPI documentation, but unfortunately, that file also contains incorrect information. So, while the auto-generator kind of works and assists in creating some models and methods, it still requires a lot of effort to complete the whole implementation.

However, we are getting there. I think part 3 (8.2.1) will involve correcting errors and adding documentation and samples. 

Any feedback and bug hunting is much appreciated.

Tolga.

**Comment by Nicelittlewood** [comment]
ToolDefinition.DefineRetrieval() generate Error when update assistant.

Tool is now: "file_search" and needed "vector_store_ids".

https://platform.openai.com/docs/assistants/tools/file-search/quickstart

```python
assistant = client.beta.assistants.update(
  assistant_id=assistant.id,
  tool_resources={"file_search": {"vector_store_ids": [vector_store.id]}},
)
```

**Comment by pappde** [comment]
I noticed the RunSteps service wasn't exposed, so submitted PR https://github.com/betalgo/openai/pull/549

In testing functions with 8.2.1 I also included some other fixes to get it working.  Those include
1) the "retrieval" tool was renamed to "file_search"
2) the RunStepResponse.StepDetails should contain MessageCreation.MessageId

I did not update the tests in the library, but I was testing this with my own code.

**Reply by Nicelittlewood** [reply]
yeah, i'm testing using my project too.

i've noticed that when i update the assistant with version 8.2.1 it generate an error for size limit of tool function description. No Issue with the 8.2.0. i had to rolleback
